pilot/pkg/model/service.go

```diff
@@ -1308,18 +1297,15 @@ func (s *Service) GetAllAddressesForProxy(node *Proxy) []string {
 func (s *Service) getAllAddressesForProxy(node *Proxy) []string {
 	if node.Metadata != nil && node.Metadata.ClusterID != "" {
 		addresses := s.ClusterVIPs.GetAddressesFor(node.Metadata.ClusterID)
-		if (features.EnableDualStack || features.EnableAmbient) && len(addresses) > 0 {
+		if (features.EnableDualStack || features.EnableAmbient) && node.GetIPMode() == Dual {
 			return addresses
 		}
 		addresses = filterAddresses(addresses, node.SupportsIPv4(), node.SupportsIPv6())
 		if len(addresses) > 0 {
 			return addresses
 		}
 	}
-	if a := s.GetAddressForProxy(node); a != "" {
-		return []string{a}
-	}
-	return nil
+	return []string{s.DefaultAddress}
```

Comment:
I am surprised that this change didn't cause failure of `TestDNSAutoAllocation`.
DNS server calls `GetAllAddressesForProxy` to build name table, and an allocated address is returned only by `GetAddressForProxy`, so I don't understand why that test passed if we no longer call `GetAddressForProxy`...?

Reply:
You are right, here i should revert and add some comments